app/src/main/kotlin/xyz/duckee/android/MainActivity.kt

```diff
@@ -109,20 +107,47 @@ class MainActivity : ComponentActivity() {
                 }
             }
         }
+
+        // Handle deeplink when the app is opened from a deeplink
+        handleDeeplink(intent)
     }
 
     override fun onResume() {
         super.onResume()
 
-        Firebase.dynamicLinks
-            .getDynamicLink(intent)
-            .addOnSuccessListener(this) { pendingDynamicLinkData: PendingDynamicLinkData? ->
-                pendingDynamicLinkData?.link?.let { deepLink ->
-                    // Handle Deeplink with Jetpack NavigationComponent
-                    navigationController?.navigate(deepLink)
-                }
+        // Handle deferred deeplink
+        Airbridge.handleDeferredDeeplink { uri ->
+            uri?.let {
+                // Handle the deferred deeplink URI
+                // For example, navigate to a specific screen based on the URI
+                handleDeeplinkUri(it)
+            }
+        }
+    }
```

Comment:
Replaced Firebase Dynamic Links handling with Airbridge's deferred deeplink handling in `onResume()`. This change allows the app to handle deferred deeplinks, which are useful for scenarios where the app wasn't installed when the user clicked the link.